kobo/hub/views.py

```diff
@@ -137,16 +132,25 @@ def _trim_log(text):
     return '<...trimmed, download required for full log>' + subtext
 
 
-def _streamed_log_response(file_path, offset, as_attachment):
+def _streamed_log_response(task, log_name, offset, as_attachment):
+    file_path = task.logs._get_absolute_log_path(log_name)
+    if not os.path.isfile(file_path) and not file_path.endswith(".gz"):
+        file_path = task.logs._get_absolute_log_path(log_name + ".gz")
+
     mimetype = mimetypes.guess_type(file_path)[0] or 'application/octet-stream'
 
     try:
         content_len = os.path.getsize(file_path) - offset
     except OSError:
         content_len = 0
 
+    try:
+        f = open(file_path, "rb")
+    except OSError:
+        return HttpResponseNotFound('Cannot find file ' + log_name)
```

Comment:
"Cannot find file" is only one possible reason an OSError might be raised.  Can we please only claim the file wasn't found if we are sure that's the case?  I think errno can be checked on the OSError, and if it's not ENOENT then we should just propagate the exception and let it end up as a 500 error rather than 404.

Code which throws away the real exception without logging it and then guesses / "lies" about the cause of an error can lead to developers wasting a lot of hours... some will blindly trust that the "Cannot find file" here really means it can't find a file and would never the consider the possibility that the error message might be false.

Reply:
Addressed in a715c03.  I've replaced `OSError` with `FileNotFoundError`.